File: week1-c/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Defining variable n as integer
    // Setting up do-while loop to force range of n to [1:8]
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1 || n > 8);

    // Left pyramid construction
    // Parameter for rows
    for (int i = 1; i < n + 1; i++)
    {
        // Paramater for columns
        for (int j = n; j > 0; j--)
        {
            // Conditional to offset the structure to the right
            if (j > i)
            {
                printf(" ");
            }
            else
            {
                printf("#");
            }
        }
        // Escape sequence to go to new line at the end of each cycle of the outer loop
        printf("\n");
    }
}

File: week6-python/sentimental-mario-more/mario.py
from cs50 import get_int

# Implement a program that prints out a double half-pyramid of a specified height

# Setting up a do-while loop situation to force user input to be in range [1:8]
n = 0
while n not in range(1, 9):
    n = get_int("Height: ")

# Nested for loop, i for rows, j and k for columns
# Two spaces between the two half-pyramids
for i in range(n):
    for j in range(n):
        if j < (n - 1 - i):
            print(" ", end="")
        else:
            print("#", end="")

    print("  ", end="")

    for k in range(n):
        if k <= i:
            print("#", end="")
            
    print()

